### PR TITLE
Add support for OIDC user info endpoint

### PR DESCRIPTION
**Pull request by geek-soft**
### JIRA link (if applicable) ###

None

### Change description ###

Adds method to client and API classes to retrieve user information from OIDC `/o/userinfo` endpoint.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/idam/client/IdamApi.java [hunk 1]
```diff
@@ -62,4 +63,11 @@ AuthenticateUserResponse authenticateUser(
     TokenExchangeResponse exchangeCode(
         @RequestBody ExchangeCodeRequest exchangeCodeRequest
     );
+
+    @GetMapping(
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/idam/client/IdamApi.java]: can't you do?
```suggestion
    @GetMapping("/o/userinfo)
```

File: src/main/java/uk/gov/hmcts/reform/idam/client/models/UserInfo.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Getter
+@Builder
+@NoArgsConstructor
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/idam/client/models/UserInfo.java]: why is this needed?

File: src/main/java/uk/gov/hmcts/reform/idam/client/IdamApi.java (hunk 1 above)
Comment (akomisarek) [reply src/main/java/uk/gov/hmcts/reform/idam/client/IdamApi.java]: I think it could be simplified like that (with "/o/userinfo") of course!

File: src/main/java/uk/gov/hmcts/reform/idam/client/models/UserInfo.java (hunk 2 above)
Comment (geek-soft) [reply src/main/java/uk/gov/hmcts/reform/idam/client/models/UserInfo.java]: Good spot, no need for that in immutable classes.

File: src/main/java/uk/gov/hmcts/reform/idam/client/IdamApi.java (hunk 1 above)
Comment (geek-soft) [reply src/main/java/uk/gov/hmcts/reform/idam/client/IdamApi.java]: Yeah, I just followed what was already there but I am happy to change it.
